[pytorch] factor out binary size upload command

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40199 [pytorch][ci] add custom selective build flow for android build
* **#40188 [pytorch] factor out binary size upload command**

Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

Differential Revision: [D22111114](https://our.internmc.facebook.com/intern/diff/D22111114)

ljk53 committed:
[pytorch] factor out binary size upload command

Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] factor out binary size upload command"


Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] factor out binary size upload command"


Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

Differential Revision: [D22111114](https://our.internmc.facebook.com/intern/diff/D22111114)

[ghstack-poisoned]

ljk53 commented on .circleci/config.yml:
```diff
@@ -169,7 +169,37 @@ commands:
+          command: |
+            docker_image="<< parameters.docker_image >>"
+            echo "docker-image: $docker_image"
+            export id=$(docker run --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -t -d -w /var/lib/jenkins ${docker_image})
```
@ezyang inspired by your comment at: https://github.com/pytorch/pytorch/pull/40199/files#r443661066 - do I need run `docker pull ${docker_image}` here explicitly? I was thinking the command runs immediately after the previous command so we can assume it runs on the same machine and can reuse the docker image locally.

Do you think we should merge this "upload_binary_size_for_android_build" command into android build command instead?

ezyang replied:
If they're in the same steps, I think all of the caches stick around.

ljk53 committed:
Update on "[pytorch] factor out binary size upload command"


Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

Differential Revision: [D22111114](https://our.internmc.facebook.com/intern/diff/D22111114)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] factor out binary size upload command"


Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

Differential Revision: [D22111114](https://our.internmc.facebook.com/intern/diff/D22111114)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] factor out binary size upload command"


Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

Differential Revision: [D22111114](https://our.internmc.facebook.com/intern/diff/D22111114)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] factor out binary size upload command"


Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

Differential Revision: [D22111114](https://our.internmc.facebook.com/intern/diff/D22111114)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] factor out binary size upload command"


Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

Differential Revision: [D22111114](https://our.internmc.facebook.com/intern/diff/D22111114)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] factor out binary size upload command"


Summary:
Create a custom command for this task to avoid copy/paste for new build jobs.

Differential Revision: [D22111114](https://our.internmc.facebook.com/intern/diff/D22111114)

[ghstack-poisoned]
